### DISCUSSION TITLE
wire:navigate preserve scroll

### DISCUSSION THREAD

**Discussion by philippbuerger** [livewire/livewire#7999] [3. Feature ideas] [answered]
It would be great to have a modifier for wire:navigate which preserves the current scroll position when the link is clicked. This should not be the default because in most situations scrolling to top after hitting a link is the expected behavior but it some cases (e.g. switching the product variant on a shop page) a scroll preserve would be the better choice. 

I read in some other discussions that this was the default behavior of a wire:navigate link? So maybe this is should not be a big deal to implement?

**Comment by michaelklopf** [comment]
Having the same problem here. A page where we have an example component in which we can navigate around. Clicking on the example items makes the view snap back to the top which is not what we want.

**Comment by digitall-it** [comment]
Did you already check this part of the documentation? [Livewire, Navigate, Preserving scroll position](https://livewire.laravel.com/docs/navigate#preserving-scroll-position)

**Reply by pdiest** [reply]
Is there a solution found? I am dealing wi with the same problem.

**Reply by firer777** [reply]
@pdiest  u can define x-data current position scroll on root element maybe html tag or use local storage session, if scroll use js update x-data or local storage session scroll position, and listen event livewire:navigated update scroll by data of x-data or local storage session scroll position

**Comment by joshhanley** [comment, marked as answer]
This has been added in PR #9341 and will be available in the next release.
